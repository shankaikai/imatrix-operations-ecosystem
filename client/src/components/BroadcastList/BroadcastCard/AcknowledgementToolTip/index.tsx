import React from "react";

interface AcknowledgementToolTipProps {
  AIFS: BroadcastRecipient[];
}

import { useState } from "react";
import {
  Popover,
  Badge,
  Image,
  Text,
  Stack,
  Group,
  Avatar,
  Indicator,
} from "@mantine/core";
import { BroadcastRecipient } from "../../../../proto/operations_ecosys_pb";

export default function AcknowledgementToolTip() {
  const [opened, setOpened] = useState(false);

  return (
    <Popover
      opened={opened}
      onClose={() => setOpened(false)}
      position="bottom"
      placement="end"
      withArrow
      trapFocus={false}
      closeOnEscape={false}
      transition="pop-top-left"
      styles={{ body: { pointerEvents: "none" } }}
      target={
        <Badge
          onMouseEnter={() => setOpened(true)}
          onMouseLeave={() => setOpened(false)}
          variant="filled"
          color="gray"
        >
          AIFS 1
        </Badge>
      }
    >
      <Stack>
        <Group position="apart">
          <Text size="sm">AIFS 1</Text>
          <Text size="sm">Location</Text>
        </Group>
        <Group>
          <Stack align="center" spacing="xs">
            <Indicator color="green" offset={5} withBorder size={14}>
              <Avatar radius="xl" />
            </Indicator>
            <Text size="xs">Name</Text>
            <Text size="xs">HP: 91119999</Text>
          </Stack>
          <Stack align="center" spacing="xs">
            <Indicator color="green" offset={5} withBorder size={14}>
              <Avatar radius="xl" />
            </Indicator>
            <Text size="xs">Name</Text>
            <Text size="xs">HP: 91119999</Text>
          </Stack>
          <Stack align="center" spacing="xs">
            <Indicator color="green" offset={5} withBorder size={14}>
              <Avatar
                radius="xl"
                src="https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=250&q=80"
              />
            </Indicator>
            <Text size="xs">Name</Text>
            <Text size="xs">HP: 91119999</Text>
          </Stack>
        </Group>
      </Stack>
    </Popover>
  );
}
