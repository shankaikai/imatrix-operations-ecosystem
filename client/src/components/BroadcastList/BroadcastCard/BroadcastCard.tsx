import {
  Badge,
  Card,
  Group,
  Text,
  Tooltip,
  useMantineTheme,
} from "@mantine/core";
import React from "react";
import { Broadcast } from "../../../proto/operations_ecosys_pb";

interface BroadcastCardProps {
  broadcast: Broadcast;
}

const mockdata = [
  {
    content: "This is a sample message",
  },
];

export default function BroadcastCard({ broadcast }: BroadcastCardProps) {
  const theme = useMantineTheme();
  const recipients = broadcast.getRecipientsList();

  return (
    <Card
      p="lg"
      shadow="sm"
      sx={{ position: "relative", borderLeft: "5px solid red" }}
    >
      <Group position="apart">
        <Text>{broadcast.getContent()}</Text>
        <Group>
          <Tooltip label={"Xavier, Emily, Hannah"}>
            <Badge variant="filled" color="gray">
              AIFS 1
            </Badge>
          </Tooltip>
          <Tooltip label={"Xavier, Emily, Hannah"}>
            <Badge variant="filled" color="gray">
              AIFS 2
            </Badge>
          </Tooltip>
          <Tooltip label={"Xavier, Emily, Hannah"}>
            <Badge variant="filled" color="green">
              AIFS 3
            </Badge>
          </Tooltip>
        </Group>
      </Group>
      <Text
        size="xs"
        color="dimmed"
        sx={{
          position: "absolute",
          right: theme.spacing.lg,
        }}
      >
        {broadcast.getCreationDate()}
      </Text>
    </Card>
  );
}
