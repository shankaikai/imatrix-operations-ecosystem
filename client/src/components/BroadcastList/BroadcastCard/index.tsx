import {
  Card,
  Group,
  Avatar,
  Stack,
  Text,
  Space,
  Badge,
  createStyles,
} from "@mantine/core";
import React from "react";
import { Checks } from "tabler-icons-react";
import truncateString from "../../../helpers/truncateString";

interface BroadcastCardProps {
  id: number;
  imgUrl: string;
  name: string;
  message?: string;
  unread?: number;
  selected?: number;
}

const useStyles = createStyles((theme) => ({}));

export default function BroadcastCard(props: BroadcastCardProps) {
  const { theme } = useStyles();

  return (
    <Card
      key={props.name}
      p="sm"
      color="gray"
      shadow="sm"
      sx={(theme) => ({
        backgroundColor:
          props.selected == props.id ? theme.colors.gray[3] : "white",
      })}
    >
      <Group position="apart">
        <Avatar src={props.imgUrl} size="lg" radius="xl" />
        <Stack sx={{ width: "240px" }}>
          <Text size="sm" weight={500}>
            {props.name}
          </Text>
          {props.message ? (
            <Group position="apart" pr="0">
              <Text size="sm" color="dimmed">
                {truncateString(props.message, 25)}
              </Text>
              {props.unread ? (
                <Badge color="red" variant="filled">
                  {props.unread}
                </Badge>
              ) : (
                <Checks color="green" />
              )}
            </Group>
          ) : (
            <Space h="sm" />
          )}
        </Stack>
      </Group>
    </Card>
  );
}
