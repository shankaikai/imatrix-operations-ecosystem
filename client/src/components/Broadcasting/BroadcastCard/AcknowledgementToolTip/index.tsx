import {
  Avatar,
  Badge,
  Group,
  Indicator,
  Popover,
  Stack,
  Text,
} from "@mantine/core";
import React, { useState } from "react";
import { AIFSRecipient } from "../../../../helpers/recipientsFormatter";

interface AcknowledgementToolTipProps {
  data: AIFSRecipient;
}

export default function AcknowledgementToolTip({
  data,
}: AcknowledgementToolTipProps) {
  const [opened, setOpened] = useState(false);

  return (
    <Popover
      opened={opened}
      onClose={() => setOpened(false)}
      position="bottom"
      placement="end"
      withArrow
      trapFocus={false}
      closeOnEscape={false}
      transition="pop-top-left"
      styles={{ body: { pointerEvents: "none" } }}
      target={
        <Badge
          onMouseEnter={() => setOpened(true)}
          onMouseLeave={() => setOpened(false)}
          variant="filled"
          color={data.allAcknowledged ? "green" : "gray"}
        >
          {data.id}
        </Badge>
      }
    >
      <Stack>
        <Group position="apart">
          <Text size="sm">{data.id}</Text>
          <Text size="sm">{data.location}</Text>
        </Group>
        <Group>
          {data.users.map((user) => (
            <Stack key={user.id} align="center" spacing="xs">
              <Indicator
                color={user.acknowledged ? "green" : "red"}
                offset={5}
                withBorder
                size={14}
              >
                <Avatar radius="xl" src={user.img} />
              </Indicator>
              <Text size="xs">{user.name}</Text>
              <Text size="xs">HP: {user.phone}</Text>
            </Stack>
          ))}
        </Group>
      </Stack>
    </Popover>
  );
}
