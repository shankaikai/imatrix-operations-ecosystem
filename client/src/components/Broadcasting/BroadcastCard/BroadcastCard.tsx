import { Card, Group, Text, useMantineTheme } from "@mantine/core";
import React from "react";
import { AIFSRecipient } from "../../../helpers/recipientsFormatter";
import AcknowledgementToolTip from "./AcknowledgementToolTip";

interface BroadcastCardProps {
  content: string;
  date: string;
  aifs: AIFSRecipient[];
}

export default function BroadcastCard({
  content,
  date,
  aifs,
}: BroadcastCardProps) {
  const theme = useMantineTheme();
  let allAcknowledged = true;
  for (var recipient of aifs) {
    allAcknowledged = allAcknowledged && recipient.allAcknowledged;
  }

  return (
    <Card
      p="lg"
      shadow="sm"
      sx={{
        position: "relative",
        borderLeft: `5px solid ${allAcknowledged ? "green" : "red"}`,
      }}
    >
      <Group position="apart">
        <Text>{content}</Text>
        <Group>
          {aifs.map((aif) => (
            <AcknowledgementToolTip key={aif.id} data={aif} />
          ))}
        </Group>
      </Group>
      <Text
        size="xs"
        color="dimmed"
        sx={{
          position: "absolute",
          right: theme.spacing.lg,
        }}
      >
        {date}
      </Text>
    </Card>
  );
}
