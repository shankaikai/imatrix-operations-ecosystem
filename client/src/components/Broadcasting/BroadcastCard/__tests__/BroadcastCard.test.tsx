import { render } from "@testing-library/react";
import {
  AIFSRecipient,
  RecipientDetails,
} from "../../../../helpers/recipientsFormatter";
import BroadcastCard from "../BroadcastCard";

const mockContent = "some content";
const mockDate = "some date";
const mockRecipientDetails: RecipientDetails[] = [
  {
    id: 1,
    phone: "some phone number",
    acknowledged: true,
    img: "some url",
    name: "some name",
  },
];

const mockData: AIFSRecipient = {
  id: "some id",
  location: "some location",
  allAcknowledged: true,
  users: mockRecipientDetails,
};
const mockAifs = [mockData];

describe("BroadcastCard", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <BroadcastCard content={mockContent} date={mockDate} aifs={mockAifs} />
    );
    expect(container).toMatchSnapshot();
  });
});
