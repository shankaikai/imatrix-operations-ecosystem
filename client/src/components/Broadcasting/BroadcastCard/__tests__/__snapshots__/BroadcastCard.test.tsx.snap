// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BroadcastCard matches snapshot 1`] = `
<div>
  <div
    class="mantine-Paper-root mantine-Card-root mantine-1671lvg"
  >
    <div
      class="mantine-Group-root mantine-19jxmdp"
    >
      <div
        class="mantine-Text-root mantine-Group-child mantine-1rxq3mb"
      >
        some content
      </div>
      <div
        class="mantine-Group-root mantine-Group-child mantine-zah578"
      >
        <div
          class="mantine-Popover-root mantine-gc2cx1"
        >
          <div
            class="mantine-1imxcuq mantine-Popover-target"
          >
            <div
              class="mantine-Badge-filled mantine-Badge-root mantine-12ynvk6"
            >
              <span
                class="mantine-h9iq4m mantine-Badge-inner"
              >
                some id
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mantine-Text-root mantine-3jfi03"
    >
      some date
    </div>
  </div>
</div>
`;
